rdrf/rdrf/testing/behaviour/features/longitudinal_followups/fh_followup_email.feature

```diff
@@ -7,75 +7,58 @@ Feature: Longitudinal followup email notifications
+    When 1 hour passes
+    Then 0 emails are sent
+
+    When 24 hours pass
+    Then 1 email is sent
+    And email 1 has 1 link
+    And each link in email 1 loads successfully
+
+    Then the mail is emptied
+
+    When 48 hours pass
+    Then no emails are sent
+
 
   Scenario: Receive a combined followup email
-    When I am logged in as curator
-    And I click "SMITH John" on patientlisting
-    And I press "Add" button in "Follow Up" group in sidebar
-    And I enter value "01-01-2023" for form "Follow Up" section " " cde "Date of assessment"
-    And I click the "Save" button
-    And I press "Add" button in "Second Follow Up" group in sidebar
-    And I enter value "01-01-2023" for form "Second Follow Up" section " " cde "Date of assessment"
-    And I click the "Save" button
-    And I send emails 1 hour later
-    Then I see 0 emails
-
-    When I send emails 24 hours later
-    Then I see 1 email
-    And I open 2 links in email 1
-    And I empty the mail
-
-    When I send emails 48 hours later
-    Then I see 0 emails
+    Given the "Follow Up" form in patient "SMITH John" is filled out
+    And the "Second Follow Up" form in patient "SMITH John" is filled out
+    When 1 hour passes
+    Then 0 emails are sent
+
+    When 24 hours pass
+    Then 1 email is sent
+    And email 1 has 2 links
+    And each link in email 1 loads successfully
+
+    Then the mail is emptied
+
+    When 48 hours pass
+    Then no emails are sent
+
 
   Scenario: Receive emails only when subscribed
-    When I am logged in as curator
-    And I click "SMITH John" on patientlisting
-    And I press "Add" button in "Follow Up" group in sidebar
-    And I enter value "01-01-2023" for form "Follow Up" section " " cde "Date of assessment"
-    And I click the "Save" button
-    And I send emails 1 hour later
-    Then I see 0 emails
-    And send emails 24 hours later
-    Then I see 1 email
-    And I empty the mail
-
-    When I click "Menu"
-    And I click "Patient List (FH Registry)"
-    And I click "SMITH John" on patientlisting
-    And I press "Add" button in "Follow Up" group in sidebar
-    And I enter value "01-01-2023" for form "Follow Up" section " " cde "Date of assessment"
-    And I click the "Save" button
-    And send emails 48 hours later
-    Then I see 1 email
-
-    When I unsubscribe in email 1
-    And I empty the mail
-    And I click "Menu"
-    And I click "Patient List (FH Registry)"
-
-    And I click "SMITH John" on patientlisting
-    And I press "Add" button in "Follow Up" group in sidebar
-    And I enter value "01-01-2023" for form "Follow Up" section " " cde "Date of assessment"
-    And I click the "Save" button
-    Then I see 0 emails
-
-    When I send emails 72 hours later
-    Then I see 0 emails
+    Given the "Follow Up" form in patient "SMITH John" is filled out
+    When 1 hour passes
+    Then 0 emails are sent
+
+    When 24 hours pass
+    Then 1 email is sent
+
+    Then the mail is emptied
```

Comment:
`Then` should only be used for assertions, so you could either cleanup the mailbox after you make the previous assertion of how many emails have been sent.. or you could change this to a `Given`, or possibly a `When`

Reply:
As I can see it, we can:

1. Change the keyword from `Then`
    a. Happy to do this, but the grammar feels a bit awkward. Can I just use `And`?
3. Add it to the links `load_step`, then make the 3rd scenario longer to test for this
    a. Adding extra steps to the scenario is the opposite of what we're going for.
5. Don't empty the emails at all, and keep a global count of 'read' emails in the outbox
    a. Keeping track of state throughout the tests makes debugging harder.

Reply:
1. If you use `And` after a `Then`, it's still part of the assertion block. But you can move it down to the `When` and make use of `And` there. 
e.g.:
```
    When 24 hours pass
    Then 1 email is sent

    When the mail is emptied
    And the "Follow Up" form in patient "SMITH John" is filled out again
    And 48 hours pass
    Then 1 email is sent
    ...
```
2. You could, but then you'd be checking `links_load` each time when you're not currently already doing that. 
3. Agree this option makes everything harder.

Reply:
Ok, I'll go with option 1